### PR TITLE
E2E: encrypt on upload, decrypt on download

### PR DESCRIPTION
**Pull request by odeke-em**
Data should only be unencrypted on the user's local disk but
should always be encrypted remotely on Google Drive, if the
password was provided.

- When a password is provided, encrypt the data before it is received
by Google Drive. Encrypt for upload:
```shell
$ drive push --encryption-password foolsGold issueFilingDrive.mp4
```

- When a password is provided, decrypt the data before writing
it to disk:
```shell
$ drive pull --piped issueFilingDrive.mp4 --decryption-password
foolsGold > txt.v
```

### CODE REVIEW

**Comment by odeke-em** [comment]
Demo here of it in action https://youtu.be/22SYefONHZk. Big shoutout to @sselph for the library.

File: src/remote.go [hunk 1]
```diff
@@ -569,6 +571,15 @@ func (r *Remote) Download(id string, exportURL string) (io.ReadCloser, error) {
 		}
 	}
 
+	if r.decrypter != nil && body != nil {
+		decR, err := r.decrypter(body)
+		if err != nil {
+			_ = body.Close()
+			return nil, err
+		}
+		body = decR
```
Comment (sselph) [inline src/remote.go]: At this point you can close the original body since the entire file is on disk.

Comment (odeke-em) [inline src/remote.go]: Thank you. Yeah I was wondering when to close it. Let me push up the fix.

**Comment by odeke-em** [comment]
Merging this in so that folks can start using it. If there are any problems we can always rollback. Also am off to work till late.

**Pull request merged**